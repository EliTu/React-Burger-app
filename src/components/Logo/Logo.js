import React from 'react';
import styles from './Logo.module.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHamburger } from '@fortawesome/free-solid-svg-icons';

const Logo = ({ size = '1x' }) => {
	// CSS Modules styles:
	const { Logo } = styles;

	return (
		<div className={Logo}>
			<FontAwesomeIcon icon={faHamburger} size={size} />
		</div>
	);
};
export default Logo;
