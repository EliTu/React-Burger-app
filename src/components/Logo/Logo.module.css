@import '../../variables.css';

.Logo {
    box-sizing: border-box;
    margin: 0.5rem 0;
    text-align: center;
    color: var(--coloring-bright-light);
}