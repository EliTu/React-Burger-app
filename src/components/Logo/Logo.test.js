import React from 'react';
import Logo from './Logo';
import { shallow } from 'enzyme';

describe('<Logo>', () => {
	const component = shallow(<Logo size="3x" />);

	it('should render the component without errors', () => {
		expect(component).toMatchSnapshot();
		expect(component.length).toBe(1);
		expect(component.length).not.toBe(2);
		expect(component.hasClass('Logo')).toBe(true);
	});

	it('should render the FontAwesome hamburger icon without errors', () => {
		let icon = component.find('FontAwesomeIcon');
		let iconType = icon.props().icon.iconName;
		let iconSize = icon.props().size;

		expect(icon.length).toBe(1);
		expect(iconType).toBe('hamburger');
		expect(iconSize).toBe('3x');
	});
});
