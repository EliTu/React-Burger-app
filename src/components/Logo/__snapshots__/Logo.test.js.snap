// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo> should render the component without errors 1`] = `
<div
  className="Logo"
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f805",
          "M464 256H48a48 48 0 0 0 0 96h416a48 48 0 0 0 0-96zm16 128H32a16 16 0 0 0-16 16v16a64 64 0 0 0 64 64h352a64 64 0 0 0 64-64v-16a16 16 0 0 0-16-16zM58.64 224h394.72c34.57 0 54.62-43.9 34.82-75.88C448 83.2 359.55 32.1 256 32c-103.54.1-192 51.2-232.18 116.11C4 180.09 24.07 224 58.64 224zM384 112a16 16 0 1 1-16 16 16 16 0 0 1 16-16zM256 80a16 16 0 1 1-16 16 16 16 0 0 1 16-16zm-128 32a16 16 0 1 1-16 16 16 16 0 0 1 16-16z",
        ],
        "iconName": "hamburger",
        "prefix": "fas",
      }
    }
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size="3x"
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
</div>
`;
